pass-import: fix plugin usage

hennersz opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

pass import needs magic to determine file types
the current python dependency used is
https://github.com/aliles/filemagic. This is very outdated and when
running the import plugin we get an error "AttributeError: module
'magic' has no attribute 'from_buffer'"
Pass import actually specifies this: https://github.com/file/file as the
required dependency. Using these python bindings fixes the error

pass-import relies on the PREFIX variable being set. Normally it is
available when the extension is called from pass because pass sources
the extension entry point script. Because of the way the extension gets
wrapped by nix we need to export PREFIX in the wrapper script.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

hennersz committed:
pass-import: use magic instead of filemagic

pass import needs magic to determine file types
the current python dependency used is
https://github.com/aliles/filemagic. This is very outdated and when
running the import plugin we get an error "AttributeError: module
'magic' has no attribute 'from_buffer'"
Pass import actually specifies this: https://github.com/file/file as the
required dependency. Using these python bindings fixes the error

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 118143` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>passExtensions.pass-import</li>
</ul>
</details>

r-rmcgibbo commented:
Result of `nixpkgs-review pr 118143` at 21980dd4 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>passExtensions.pass-import</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/tools/security/pass/extensions/import.nix:25:15:
```
   |
25 |   patches = [ ./0001-Fix-installation-with-Nix.patch ];
   |               ^
```
</li>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/tools/security/pass/extensions/import.nix:1:62:
```
  |
1 | { lib, stdenv, fetchFromGitHub, pythonPackages, makeWrapper, fetchpatch }:
  |                                                              ^
```
</li>
  </ul>
</details>

hennersz committed:
pass-import: export PREFIX variable in wrapper

pass-import relies on the PREFIX variable being set. Normally it is
available when the extension is called from pass because pass sources
the extension entry point script. Because of the way the extension gets
wrapped by nix we need to export PREFIX in the wrapper script.

SuperSandro2000 commented:
@ofborg eval

hennersz commented:
closing due to changes made in https://github.com/NixOS/nixpkgs/pull/117165
